infrastructure-playbooks/filestore-to-bluestore.yml

```diff
@@ -25,34 +28,55 @@
     - import_role:
         name: ceph-defaults
 
-    - name: set_fact current_objectstore
-      set_fact:
-        current_objectstore: '{{ osd_objectstore }}'
+    - name: import_role ceph-facts
+      import_role:
+        name: ceph-facts
+        tasks_from: container_binary.yml
 
-    - name: warn user about osd already using bluestore
-      debug:
-        msg: 'WARNING: {{ inventory_hostname }} is already using bluestore. Skipping all tasks.'
-      when: current_objectstore == 'bluestore'
+    - name: set_fact container_run_cmd, container_exec_cmd
+      set_fact:
+        container_run_cmd: "{{ container_binary + ' run --rm --privileged=true --net=host --pid=host --ipc=host -v /dev:/dev -v /etc/ceph:/etc/ceph -v /var/lib/ceph:/var/lib/ceph -v /var/run:/var/run --entrypoint=' if containerized_deployment | bool else '' }}ceph-volume {{ ceph_docker_registry + '/' + ceph_docker_image + ':' + ceph_docker_image_tag if containerized_deployment | bool else '' }}"
+        container_exec_cmd: "{{ container_binary + ' exec ceph-mon-' + hostvars[groups[mon_group_name][0]]['ansible_hostname'] if containerized_deployment | bool else '' }}"
+
+    - name: get ceph osd tree data
+      command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} osd tree -f json"
+      delegate_to: "{{ groups[mon_group_name][0] }}"
+      register: osd_tree
+      changed_when: false
+      run_once: true
 
-    - name: shrink and redeploy filestore osds
-      when: current_objectstore == 'filestore'
-      block:
-        - import_role:
-            name: ceph-facts
-            tasks_from: container_binary.yml
+    - name: set_fact osd_ids
+      set_fact:
+        osd_ids: "{{ osd_ids | default([]) + [item] }}"
```

Comment:
[item] -> item as it's already a list otherwise you will have list in a list

"children":[0,1,2]